gapis/api/vulkan/command_buffer_rebuilder.go

```diff
@@ -24,14 +24,14 @@ import (
 	"github.com/google/gapid/gapis/memory"
 )
 
-// unpackMap takes a dense map of u32 -> structure, flattens the map into
-// a slice, allocates the appropriate data and returns it as well as the
-// length of the map.
-func unpackMap(ctx context.Context, s *api.GlobalState, m interface{}) (api.AllocResult, uint32) {
+// unpackMapCustom takes a dense map of u32 -> structure, flattens the map into
+// a slice, allocates the appropriate data using a custom provided allocation
+// function and returns it as well as the length of the map.
+func unpackMapCustom(alloc func(v ...interface{}) api.AllocResult, m interface{}) (api.AllocResult, uint32) {
```

Comment:
Maybe `unpackMapWithAllocator` it describes what's custom about it a bit more clearly.